### PR TITLE
Remove dependency on flask_swagger and improve spec

### PR DESCRIPTION
**Pull request by Glaisher**

### CODE REVIEW

File: spec.v1.yml [hunk 1]
```diff
@@ -0,0 +1,364 @@
+openapi: 3.0.0
+info:
+  version: "1.0"
+  title: Sunnah.com API
+  description: >-
+    Sunnah.com API offers access to verified data of collections of hadith of
+    the Prophet (SAWS). Here you will find a list of the available endpoints in
+    our REST API.
+
+
```
Comment (hasankhan) [inline spec.v1.yml:10]: remove extra empty lines?

File: spec.v1.yml [hunk 2]
```diff
@@ -189,26 +166,49 @@ paths:
...
       parameters:
         - in: path
           name: collectionName
           description: name of collection
           required: true
           schema:
             type: string
+        - in: path
+          name: hadithNumber
+          description: number of the hadith
```
Comment (ahadith) [inline spec.v1.yml:197]: Please capitalize the first letter in the `summary` and `description` fields (not just here but consistently throughout the file). Also, lowercase Collection and Book when they occur in the middle of a sentence in those fields.

File: spec.v1.yml (hunk 1 above)
Comment (Glaisher) [reply spec.v1.yml:10]: Looks like we need it to have the line break appear in swagger UI, if removing the newline it disappears.

**Comment by Glaisher** [comment]
The spec file can be directly downloaded from api-docs so I think it redundant for us to serve it again.

File: spec.v1.yml [hunk 3]
```diff
@@ -2,9 +2,29 @@ swagger: '2.0'
 info:
   version: '1.0'
   title: Sunnah.com API
+  description: |-
+    Sunnah.com API offers access to verified data of collections of hadith of the Prophet (SAWS). Here you will find a list of the available endpoints in our REST API.
+
+    You will need an API key to access this data, which you may request from https://sunnah.com/contact.
```
Comment (ahadith) [inline spec.v1.yml:11]: Can we instead have them open an issue on GitHub with some basic required information that we specify in the README? (name, email, purpose, rate requested)

File: spec.v1.yml [hunk 4]
```diff
@@ -2,9 +2,29 @@ swagger: '2.0'
...
+  contact:
+    name: Sunnah.com
+    url: 'https://sunnah.com/contact'
+    email: sunnah@iman.net
```
Comment (ahadith) [inline spec.v1.yml:15]: I would rather not include an email address if that's possible at all.

File: spec.v1.yml [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+      in: query
+      name: page
+      description: Offset for pagination
+      required: false
+      schema:
+        type: integer
+        default: 1
+  securitySchemes:
+    APIKey:
+      name: X-API-Key
```
Comment (Ysajid) [inline spec.v1.yml:235]: I think the current implementation looks for the `x-aws-secret` value in the header.

Comment (ahadith) [reply spec.v1.yml:235]: No, that's a secret between API Gateway and our server. `X-API-Key` is what API consumers include in their request.

File: spec.v1.yml (hunk 4 above)
Comment (Glaisher) [reply spec.v1.yml]: Removed

File: spec.v1.yml (hunk 3 above)
Comment (Glaisher) [reply spec.v1.yml]: Added issue template for this

File: spec.v1.yml (hunk 2 above)
Comment (Glaisher) [reply spec.v1.yml]: Done

**Comment by Glaisher** [comment]
Addressed the comments

File: spec.v1.yml [hunk 6]
```diff
@@ -258,15 +258,15 @@ components:
...
         bookNumber:
           type: string
           description: The number of the book this hadith belongs to
         chapterId:
           type: string
-          description: The id of the chapter this hadith belongs to
+          description: The ID of the chapter this hadith belongs to
         hadithNumber:
-          type: integer
+          type: string
```
Comment (ahadith) [inline spec.v1.yml:269]: So from the DB, this is a decimal format with 2 decimal places. Are we treating it as a string consistently as far as its exposure to consumer is concerned? (i.e. in wherever it is being exposed?)

Comment (Glaisher) [reply spec.v1.yml:269]: `hadithNumber` is a varchar in the `HadithTable` and also there are values such as `100 b` so string is correct, I believe.

**Pull request merged**